### ISSUE TITLE
parse error, jekyll-import gem from tumblr

### ISSUE THREAD

**Issue by mattkhoury** [jekyll/jekyll-import#252]
Trying to import my Tumblr site to Jekyll using the jekyll-import gem. Obviously, I'm not having much success. Any have any idea how to go about troubleshooting this error? Oh, and I'm new to Ruby - and - Jekyll, so, learning as I go.

`(JSON::ParserError)
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/json/common.rb:155:in `parse'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-import-0.10.0/lib/jekyll-import/importers/tumblr.rb:45:in `process'
	from /Library/Ruby/Gems/2.0.0/gems/jekyll-import-0.10.0/lib/jekyll-import/importer.rb:23:in `run'
	from -e:2:in `<main>'`

**Comment by parkr** [comment]
Hey! Looking through our changelog, #243 and #249 may have fixed this. The error you're seeing is from malformed JSON. I can release a new version on Monday with the updated code. In the meantime, you can modify the code locally to pull in the JSON from the API as we do in #249. One way to do this is to copy the contents of tumblr.rb from GitHub here to your local gem (path is pasted above). Another way is to wait until Monday :-)

**Comment by mattkhoury** [comment]
Hi Parkr, I'll give it another shot today. Sorry for the delay responding to you.

**Comment by hairyhenderson** [comment]
This just happened to me, so I tried applying the fix from #249 into `tumblr.rb`, but there's an additional bug there that causes this:

```
Fetching http://myblog.tumblr.com/api/read/json/?num=50&start=0
/Users/me/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/json/common.rb:155:in `initialize': no implicit conversion of nil into String (TypeError)
	from /Users/me/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/json/common.rb:155:in `new'
	from /Users/me/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/json/common.rb:155:in `parse'
	from /Users/me/.rvm/gems/ruby-2.2.1/gems/jekyll-import-0.10.0/lib/jekyll-import/importers/tumblr.rb:48:in `process'
	from /Users/me/.rvm/gems/ruby-2.2.1/gems/jekyll-import-0.10.0/lib/jekyll-import/importer.rb:23:in `run'
	from -e:2:in `<main>'
```

This is because in this block:
```rb
          feed = open(feed_url)
          contents = feed.readlines.join("\n")
          beginning = contents.index("{")
          ending = contents.rindex("}")
          json = feed.readlines.join("\n")[beginning...ending]  # Strip Tumblr's JSONP chars.
```

Looks like the last line _should_ be `json = contents[beginning...ending]` - the stream's already been read so `feed.readlines` returns an empty string.

After fixing this, though, I get another parse error:

```
/Users/me/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/json/common.rb:155:in `parse': 757: unexpected token at '<a whole lotta JSON>'  (JSON::ParserError)
	from /Users/me/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/json/common.rb:155:in `parse'
	from /Users/me/.rvm/gems/ruby-2.2.1/gems/jekyll-import-0.10.0/lib/jekyll-import/importers/tumblr.rb:48:in `process'
	from /Users/me/.rvm/gems/ruby-2.2.1/gems/jekyll-import-0.10.0/lib/jekyll-import/importer.rb:23:in `run'
	from -e:2:in `<main>'
```

Looks like `ending = contents.rindex("}")` is also wrong - needs to be `ending = contents.rindex("}") + 1`.

After fixing that, things seem to be churning away.

I may issue a PR with these fixes... :wink:
